src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBResource.cs

```diff
@@ -125,4 +124,35 @@ public override void AddRoleAssignments(IAddRoleAssignmentsContext roleAssignmen
 
         AzureCosmosExtensions.AddContributorRoleAssignment(infra, cosmosAccount, principalId);
     }
+
+    internal ReferenceExpression GetConnectionString(string? databaseName = null, string? containerName = null)
```

Comment:
Consider just using this to always built the connection string. Not just using it for child resources.